### PR TITLE
feat: managing custom accounting rule linked setting

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Review by tewshi — APPROVED** [review]
lgtm!

File: frontend/app/src/types/settings/accounting.ts [hunk 1]
```diff
@@ -57,3 +51,9 @@ export interface AccountingRuleRequestPayload
   readonly eventSubtypes?: string[];
   readonly counterparties?: string[];
 }
+
+export interface AccountingRuleLinkedSettingMap {
+  identifier: string;
+  label: string;
+  state: Ref<any>;
```
Comment (kelsos) [inline frontend/app/src/types/settings/accounting.ts:58]: Can we do something about this any here? the type should not be any right? can we infer the proper type?

File: frontend/app/src/composables/settings/accounting/rule-mapping.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        if (state) {
+          const label = te(translationKey)
+            ? t(translationKey)
+            : toHumanReadable(item);
+
+          result.push({
+            identifier: item,
+            label,
+            state
```
Comment (kelsos) [inline frontend/app/src/composables/settings/accounting/rule-mapping.ts:39]: Do you use this to change the setting then? this is why you pass the ref here?

Comment (kelsos) [reply frontend/app/src/composables/settings/accounting/rule-mapping.ts:39]: I am not sure I like that we return a ref from a computed.

File: frontend/app/src/components/settings/accounting/rule/AccountingRuleWithLinkedSettingDisplay.vue [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  }
+
+  return null;
+});
+
+const value: ComputedRef<boolean> = computed(() => {
+  const selectedLinkableSettingVal = get(selectedLinkableSetting);
+  if (selectedLinkableSettingVal) {
+    return get(get(selectedLinkableSettingVal).state);
```
Comment (kelsos) [inline frontend/app/src/components/settings/accounting/rule/AccountingRuleWithLinkedSettingDisplay.vue:37]: We should not need to do a get-get, the computed property should update any time any of the state updates if we unref inside right? does it not?

File: frontend/app/src/components/settings/accounting/rule/AccountingRuleSetting.vue [hunk 4]
```diff
@@ -149,11 +152,17 @@ const showDeleteConfirmation = (item: AccountingRuleEntry) => {
...
+  getEventTypeData({
+    eventType,
+    eventSubtype
+  });
 </script>
 
 <template>
   <div>
-    <TablePageLayout :title="[t('accounting_settings.rule.title')]">
+    <TablePageLayout child :title="[t('accounting_settings.rule.title')]">
```
Comment (kelsos) [inline frontend/app/src/components/settings/accounting/rule/AccountingRuleSetting.vue:165]: Can we make the `TablePageLayout` here the root element and then move the form in it's default slot to remove the extra div?

**Review by kelsos — APPROVED** [review]
lgtm

**Pull request merged**